scripts/idl/generators/cpp/BridgeClustersCpp.jinja

```diff
@@ -0,0 +1,240 @@
+#include <app-common/zap-generated/cluster-id.h>
+#include <app-common/zap-generated/attribute-id.h>
+
+#include <new>
+
+namespace clusters {
+{% for struct in structs %}
+struct {{struct.name}}
+{
+  CHIP_ERROR Decode(chip::TLV::TLVReader & reader)
+  {
+    chip::app::Clusters::detail::Structs::{{struct.name}}::DecodableType t;
+    CHIP_ERROR err = t.Decode(reader);
+    if(err == CHIP_NO_ERROR) {
+      {%-      for field in struct.fields %}
+      {{field.name}} = t.{{field.name}};
+      {%-      endfor  %}
+    }
+    return err;
+  }
+
+  CHIP_ERROR Encode(chip::TLV::TLVWriter & writer, chip::TLV::Tag tag) const
+  {
+    chip::app::Clusters::detail::Structs::{{struct.name}}::Type t;
+    {%-      for field in struct.fields %}
+    t.{{field.name}} = {{field.name}};
+    {%-      endfor  %}
+    return t.Encode(writer, tag);
+  }
+
+  {%-   for field in struct.fields %}
+  {{field | getField(None, idl)}} {{field.name}};
+  {%-  endfor  %}
+};
+{%- endfor %}
+
+{%- for cluster in clusters %}
+{%-   if cluster | dynamicCluster(idl) %}
+struct {{cluster.name}}Cluster : public CommonCluster
```

Comment:
Is this generating a header  only? Could we split generating a header and cpp file?

Reply:
That gets tricky as the generated code depends on classes declared by the bridge app, so moving things around in the bridge app could mean changing the codegen to output new include paths. If it stays header-only then the only requirement is that these structs be declared somewhere prior to inclusion

Reply:
Could you explain with an example? trying to wrap my head around the statement of 'depends on classes declared by bridge app'. Is this a circular include case?

Reply:
Not circular includes no, but the Attribute template is defined in bridge-app/linux/include/Clusters.h and is used in the generated code, so any generated file either has to include that, or get included after that header

Reply:
The generated code should be including the `bridge-app/linux/include/Clusters.h`.